### DISCUSSION TITLE
Mirth Alert on Data Value in Segment

### DISCUSSION THREAD

**Discussion by IDAccessGoddess** [nextgenhealthcare/connect#4671] [Q&A]
Hey all, I'm looking to try and set up an alert on a channel if a message comes through with a certain data value in the PID segment. Is this possible?

**Comment by pacmano1** [comment]
Yes, what have you tried?   The forums has lots to read about on alerting when a condition is met.

**Reply by IDAccessGoddess** [reply]
I must be searching wrong because I couldn't find anything of value. I tried the general alerts but I couldn't narrow it down. 
I went to the channel transformers and tried to trigger an alert that way but I wasn't sure where to put it, or if that's even the correct place to put it.

**Reply by pacmano1** [reply]
You can generate an alert virtually anywhere that you can put javascript in Mirth.

In the forum search box use "send alerts".

See https://forums.mirthproject.io/forum/mirth-connect/support/17492-mirth-connect-events-by-email
See https://forums.mirthproject.io/forum/mirth-connect/support/17445-send-an-email-alert-when-a-message-get-filtered-at-source-filter

**Reply by IDAccessGoddess** [reply]
So the regex I want to use involves part of a word, not the whole word. Will it still pick that up?

**Reply by pacmano1** [reply]
Yes.  If you do not post any code, it is nearly impossible to help you.

**Reply by tonygermano** [reply]
Until https://github.com/nextgenhealthcare/connect/pull/4408 is accepted, there is a bug where manually sending an alert from a channel will not populate the messageId variable. If you need that information, you will need to include it in the alert message itself.
